stream_alert/threat_intel_downloader/threat_stream.py

```diff
@@ -0,0 +1,260 @@
+        Args:
+            event (dict): Contains lambda function invocation information. Initially,
+                Threat Intel Downloader lambda funciton is invoked by Cloudwatch
+                event. 'next_url' key will be inserted to event lambda function
+                invokes itself to retrieve more IOCs.
+
+        Returns:
+            (tuple): (list, str, bool)
+                - First object is a list of intelligence.
+                - Second object is a string of next token to retrieve more IOCs.
+                - Third object is bool to indicated if retrieve more IOCs from
+                    threat feed.
+        """
+        if not event:
+            return None, None, False
+
+        next_url = None
+        if 'next_url' not in event:
+            next_url = '/api/v2/{}/?username={}&api_key={}&status={}&limit={}'\
+                .format(self._API_RESOURCE,
+                        self.api_user,
+                        self.api_key,
+                        self._IOC_STATUS,
+                        self._API_MAX_LIMIT)
+        else:
+            next_url = event.get('next_url', None)
+
+        if not next_url:
+            return None, None, False
+
+        return self._connect(next_url)
+
+    def _process_data(self, data):
+        """Process and filter data by sources and keys
+        Args:
+            data (list): A list contains ioc information
+                Example:
+                    [
+                        {
+                            'value': 'malicious_domain.com',
+                            'itype': 'c2_domain',
+                            'source': 'crowdstrike',
+                            'type': 'domain',
+                            'expiration_ts': '2017-12-19T04:45:18.412Z',
+                            'key1': 'value1',
+                            'key2': 'value2',
+                            ...
+                        },
+                        {
+                            'value': 'malicious_domain2.com',
+                            'itype': 'c2_domain',
+                            'source': 'ioc_source2',
+                            'type': 'domain',
+                            'expiration_ts': '2017-12-31T04:45:18.412Z',
+                            'key1': 'value1',
+                            'key2': 'value2',
+                            ...
+                        }
+                    ]
+
+        Returns:
+            (list): A list of dict contains useful IOC information
+                Example:
+                    [
+                        {
+                            'value': 'malicious_domain.com',
+                            'itype': 'c2_domain',
+                            'source': 'crowdstrike',
+                            'type': 'domain',
+                            'expiration_ts': 1513658718,
+                        }
+                    ]
+        """
+        def _epoch_time(time_str):
```

Comment:
making this is `@staticmethod` instead of a nested function would make testing this much easier. I'd avoid nested functions if possible. Currently there is no way to reference this function from outside of its containing function, so testing it properly becoming inherently difficult.